plugins/app-backend/src/service/router.ts

```diff
@@ -123,26 +134,162 @@ export async function createRouter(
 
   logger.info(`Serving static app content from ${appDistDir}`);
 
-  let injectedConfigPath: string | undefined;
-  if (!disableConfigInjection) {
-    const appConfigs = await readConfigs({
-      config,
-      appDistDir,
-      env: process.env,
-      schema: options.schema,
-    });
+  const appConfigs = disableConfigInjection
+    ? undefined
+    : await readConfigs({
+        config,
+        appDistDir,
+        env: process.env,
+      });
 
-    injectedConfigPath = await injectConfig({ appConfigs, logger, staticDir });
-  }
+  const assetStore =
+    options.database && !disableStaticFallbackCache
+      ? await StaticAssetsStore.create({
+          logger,
+          database: options.database,
+        })
+      : undefined;
 
   const router = Router();
 
   router.use(helmet.frameguard({ action: 'deny' }));
 
+  const publicDistDir = resolvePath(appDistDir, 'public');
+
+  const enablePublicEntryPoint =
+    (await fs.pathExists(publicDistDir)) && auth && httpAuth;
+
+  if (enablePublicEntryPoint && auth && httpAuth) {
```

Comment:
```suggestion
  if (enablePublicEntryPoint) {
```

already checked above ... if typescript requires it to be happy, then maybe just remove the checks on the lines above instead

Reply:
It's to make TypeScript happy, and it needs to stack in the check above to make sure the usage further down is consistent